addons/github/api.py

```diff
@@ -151,7 +151,32 @@ def revoke_token(self):
             return self.gh3.revoke_authorization(self.access_token)
 
     def check_authorization(self):
-        return self.gh3.check_authorization(self.access_token)
+        """Check an authorization created by a registered application.
+
+        OAuth applications can use this method to check token validity
+        without hitting normal rate limits because of failed login attempts.
+        If the token is valid, it will return True, otherwise it will return
+        False.
+
+        :returns:
+            True if token is valid, False otherwise
+        :rtype:
+            bool
+        """
+        p = self.gh3.session.params
+        auth = (p.get('client_id'), p.get('client_secret'))
+        if self.access_token and auth:
+            url = self.gh3._build_url('user')
+            resp = self.gh3._get(
+                url,
+                headers={
+                    'Authorization': f'token {self.access_token}',
+                    'Accept': 'application/vnd.github.v3+json'
+                },
+                params={'client_id': p.get('client_id'), 'client_secret': p.get('client_secret')},
```

Comment:
Are these params required for this to work, or does just accessing with the token auth header do all the work here?

Reply:
Appears to only require the `Authorization: token <token>` header

Reply:
Yep I confirmed it locally